ansible/ocs.yaml

```diff
@@ -0,0 +1,174 @@
+      shell: oc get pods -n local-storage
+      retries: 100
+      delay: 20
+      register: local_storage_operator_ready
+      until: (local_storage_operator_ready.stdout | regex_findall('local-storage-operator-.+-.+Running') | length) == 1
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Create local-storage PVs
+      shell: |
+        oc apply -f {{ working_yamls_dir }}/ocs/local-storage-volumes.yaml
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Deploy OCS subscription
+      shell: oc apply -f {{ working_yamls_dir }}/ocs/ocs-sub.yaml
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Wait for OCS, rook-ceph and noobaa operators to be ready
+      shell: oc get pods -n openshift-storage
+      retries: 100
+      delay: 20
+      register: ocs_operators_ready
+      until: (ocs_operators_ready.stdout | regex_findall('ocs-operator-.+-.+Running') | length) == 1 and (ocs_operators_ready.stdout | regex_findall('rook-ceph-operator-.+-.+Running') | length) == 1 and (ocs_operators_ready.stdout | regex_findall('noobaa-operator-.+-.+Running') | length) == 1
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Deploy OCS cluster
+      shell: oc apply -f {{ working_yamls_dir }}/ocs/ocs-storage-cluster.yaml
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Wait for OCS pods to be ready
+      shell: oc get pods -n openshift-storage
+      retries: 100
+      delay: 30
+      register: ocs_pods_ready
+      until: ((ocs_pods_ready.stdout | regex_findall('csi-cephfsplugin-.+Running') | length) == (ocs_pods_ready.stdout | regex_findall('csi-cephfsplugin-.+') | length) and (ocs_pods_ready.stdout | regex_findall('csi-cephfsplugin-.+Running') | length) != 0)
+             and ((ocs_pods_ready.stdout | regex_findall('csi-rbdplugin-.+Running') | length) == (ocs_pods_ready.stdout | regex_findall('csi-rbdplugin-.+') | length) and (ocs_pods_ready.stdout | regex_findall('csi-rbdplugin-.+Running') | length) != 0)
+             and ((ocs_pods_ready.stdout | regex_findall('rook-ceph-mon-.+Running') | length) == (ocs_pods_ready.stdout | regex_findall('rook-ceph-mon-.+') | length) and (ocs_pods_ready.stdout | regex_findall('rook-ceph-mon-.+Running') | length) != 0)
+             and ((ocs_pods_ready.stdout | regex_findall('rook-ceph-osd-\d-.+Running') | length) == (ocs_pods_ready.stdout | regex_findall('rook-ceph-osd-\d-.+') | length) and (ocs_pods_ready.stdout | regex_findall('rook-ceph-osd-\d-.+Running') | length) != 0)
+             and ((ocs_pods_ready.stdout | regex_findall('noobaa-core-0.+Running') | length) == 1)
+             and ((ocs_pods_ready.stdout | regex_findall('noobaa-db-0.+Running') | length) == 1)
```

Comment:
is there not a higher level CR that rolls up the status for all of these into a single condition we could wait on? This is totally fine if it works, just wondering as this seems like it could depend on internal pod names that could change

Reply:
There is a higher-level CR, the `StorageCluster`, which spawns all of the pods that we look for in this task.  The problem is that it initially reports `Ready` even before these pods are created:

```
[root@bkr-hv01 ~]# oc get storagecluster -n openshift-storage
NAME                 AGE   PHASE   EXTERNAL   CREATED AT             VERSION
ocs-storagecluster   4s    Ready              2020-09-24T11:34:32Z   4.5.0
```

It eventually goes into the `Progressing` phase and never leaves that, even after Ceph is up and ready for use.  I think this is due to the minimal footprint settings, as some pods can't get scheduled to the chosen storage node due to resource constraints:

```
...
rook-ceph-mds-ocs-storagecluster-cephfilesystem-a-54f8585fwb5s4   1/1     Running     0          107s    10.128.4.19      worker-4   <none>           <none>
rook-ceph-mds-ocs-storagecluster-cephfilesystem-b-c4844bd5255hh   0/1     Pending     0          106s    <none>           <none>     <none>           <none>
...
```

It might be possible to rely on the phase of the `StorageCluster` resource if we up the memory/CPU on the storage node to allow all pods to be scheduled.  This playbook just currently follows all the recommended settings for one-node OCS.

Reply:
@abays thanks for clarifying.

Maybe just add a comment that says something like "FIXME: investigate replacing this with a wait_for StorageCluster condition if it is supported"

Reply:
Ack, will do